src/ip_voter.cpp

```diff
@@ -173,14 +173,28 @@ namespace libtorrent
 		return true;
 	}
 
-	bool external_ip::cast_vote(address const& ip, int source_type, address const& source)
+	external_ip::external_ip(address local4, address global4
```

Comment:
any reason not to use `const&` for addresses? (IPv6 are a little heavier I think)